src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7533,9 +7533,21 @@ private StatementSyntax ParseEmbeddedStatement(bool complexCheck)
                 case SyntaxKind.LabeledStatement:
                 case SyntaxKind.LocalDeclarationStatement:
                 case SyntaxKind.LocalFunctionStatement:
-                case SyntaxKind.DeconstructionDeclarationStatement:
                     statement = this.AddError(statement, ErrorCode.ERR_BadEmbeddedStmt);
                     break;
+                case SyntaxKind.ExpressionStatement:
```

Comment:
I think we should get guidance from Mads as to whether or not an error is deserved in this situation. I thought that it isn't considered a declaration any longer (from the spec point of view) due to the syntax change.

Whichever way you leave this code, please open a separate issue to resolve this.